custom-routes-vis/src/main/webapp/script.js

```diff
@@ -201,95 +205,130 @@ function generateRoutes() {
 
 /**
  * Creates a polyline for a route on the map.
- * @param {num} routeNum The index of the route in the order as it is returned
- * from the API.
- * @param {!JSON} routeJson JSON object containing all information of the target
+ * @param {num} routeNum The index of the route in the order that it is returned
+ * from the selected API.
+ * @param {!Object} routeJson JSON object containing all information of the target
  * route.
- * @param {bool} directionsApi Whether the route is obtained by the Directions
+ * @param {boolean} directionsApi Whether the route is obtained by the Directions
  * API; if false, assume it is obtained by the Routes Preferred API.
  */
 function createRoutePolyline(routeNum, routeJson, directionsApi) {
   let routeLegs = routeJson.legs;
   let routeCoordinates = [];
 
   // Total duration of route in seconds.
-  let totalDuration = directionsApi ? 0 : routeJson.duration;
+  let totalDurationSec = directionsApi ? 0 : routeJson.duration;
   // Total distance of route in meters.
-  let totalDistance = directionsApi ? 0 : routeJson.distanceMeters;
+  let totalDistanceMeters = directionsApi ? 0 : routeJson.distanceMeters;
   // Note: Routes Preferred has duration and distanceMeters attributes for each
-  // route, but Directions only has those for each leg of the route.
+  // route, but Directions only has them for each leg of the route.
 
+  // Get all coordinates given by the steps of the legs of the route.
   for (let i = 0; i < routeLegs.length; i++) {
     let legSteps = routeLegs[i].steps;
     console.log('num steps', legSteps.length);
 
     for (let j = 0; j < legSteps.length; j++) {
-      if (directionsApi) {
-        routeCoordinates.push(legSteps[j].start_location);
-      } else {
-        routeCoordinates.push(legSteps[j].startLocation.LatLng);
-      }
+      directionsApi ?
+          routeCoordinates.push(legSteps[j].start_location) :
+          routeCoordinates.push(legSteps[j].startLocation.LatLng);
```

Comment:
What i meant was : `routeCoordinates.push(directionsApi ? legSteps[j].start_location : legSteps[j].startLocation.LatLng)`

The goal is to not repeat the common part.

```suggestion
      directionsApi ?
          routeCoordinates.push(legSteps[j].start_location) :
          routeCoordinates.push(legSteps[j].startLocation.LatLng);
```

Reply:
Same for the places below.

Reply:
Ah got it. This part is deleted though